src/main/java/uk/gov/pay/webhooks/message/WebhookMessageService.java

```diff
@@ -64,17 +65,30 @@ public void handleInternalEvent(InternalEvent event) throws IOException, Interru
             subscribedWebhooks
                     .stream()
                     .map(webhook -> buildWebhookMessage(webhook, event, ledgerTransaction))
+                    .flatMap(Optional::stream)
                     .map(webhookMessageDao::create)
                     .forEach(webhookMessageEntity -> webhookDeliveryQueueDao.enqueueFrom(webhookMessageEntity, WebhookDeliveryQueueEntity.DeliveryStatus.PENDING, Date.from(instantSource.instant())));
         }
     }
 
-    private WebhookMessageEntity buildWebhookMessage(WebhookEntity webhook, InternalEvent event, LedgerTransaction ledgerTransaction) {
-        JsonNode resource = switch (event.resourceType()) {
-            case "payment" -> objectMapper.valueToTree(PaymentApiRepresentation.of(ledgerTransaction));
-            case "refund" -> objectMapper.valueToTree(RefundApiRepresentation.of(ledgerTransaction));
-            default -> objectMapper.valueToTree(ledgerTransaction);
+    private Optional<WebhookMessageEntity> buildWebhookMessage(WebhookEntity webhook, InternalEvent event, LedgerTransaction ledgerTransaction) {
+        return switch (event.resourceType()) {
+            case "payment" -> {
+                var resource = objectMapper.valueToTree(PaymentApiRepresentation.of(ledgerTransaction));
+                yield Optional.of(buildWebhookMessageEntity(webhook, event, resource));
```

Comment:
First use of `yield` in the Pay codebase!

(It was almost `break`. Yes, you’d write `break Optional.of(buildWebhookMessageEntity(webhook, event, resource));`!)